extensions/google-translate/src/actions.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { Action, ActionPanel, getPreferenceValues } from "@raycast/api";
+
+enum DefaultActionPreference {
+  CopyToClipboard = "copy",
+  PasteToApp = "paste",
+}
+
+interface Preferences {
+  defaultAction?: DefaultActionPreference;
+}
+
+interface ActionsOpts {
+  value: string;
+  defaultActionsPrefix?: string;
+  firstSectionTitle?: string;
+  otherActions?: React.ReactElement[];
+  otherSections?: React.ReactElement[];
+}
+
+export function getActions({
+  value,
+  defaultActionsPrefix,
+  firstSectionTitle,
+  otherActions,
+  otherSections,
+}: ActionsOpts) {
+  const defaultPreference = getPreferenceValues<Preferences>().defaultAction;
+  const action = [
```

Comment:
Factories like this is not good in react, let's use it as a React component, and create this as `ConfigurableAction`, and use it in the code as usual, instead of returning whole Panel component from this function and put any configurations to this factory method

If you are not so familiar with react, I can do that refactoring in separate PR

Reply:
It can be simply made like this:

```tsx
const ConfigurableAction = () => {
  const defaultPreference = getPreferenceValues<Preferences>().defaultAction
  
  if (defaultPreference === DefaultActionPreference.CopyToClipboard) {
    return (
      <Action.CopyToClipboard
        title={defaultActionsPrefix ? `Copy ${defaultActionsPrefix}` : `Copy`}
        key={DefaultActionPreference.CopyToClipboard}
        content={value}
      />
    );
  }

  if (defaultPreference === DefaultActionPreference.PasteToApp) {
    return (
      <Action.Paste
        title={defaultActionsPrefix ? `Paste ${defaultActionsPrefix}` : `Paste`}
        key={DefaultActionPreference.PasteToApp}
        content={value}
      />
    );
  }

  return null;
};
```

Reply:
it will simplify this change a lot, and will be more flexible, this component will be pretty reusable over the app, and will be easier to support. Will also not break the React structure

Reply:
I've made a PR to your Repo, can you please check it? https://github.com/Popalay/raycast-extensions/pull/1

If you will merge it, the changes will go right to this PR, and I think we will be good to merge it